telemetry/config.go

```diff
@@ -0,0 +1,254 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/host"
+	"go.opentelemetry.io/contrib/instrumentation/runtime"
+	otelconf "go.opentelemetry.io/contrib/otelconf/v0.3.0"
+	"go.opentelemetry.io/contrib/propagators/b3"
+	"go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	logglobal "go.opentelemetry.io/otel/log/global"
+	lognoop "go.opentelemetry.io/otel/log/noop"
+	metricnoop "go.opentelemetry.io/otel/metric/noop"
+	"go.opentelemetry.io/otel/propagation"
+	logsdk "go.opentelemetry.io/otel/sdk/log"
+	metricsdk "go.opentelemetry.io/otel/sdk/metric"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	tracenoop "go.opentelemetry.io/otel/trace/noop"
+	"go.yaml.in/yaml/v3"
+)
+
+const (
+	OtelConfigEnvVar = "OTEL_EXPERIMENTAL_CONFIG_FILE"
```

Comment:
We talked previously about how using init avoided atomic reference checks via the delegation field in otel, but I do agree that it would be nice to keep things more idiomatic.

Is there no way to parse the node dir passed via app config or cli and assume that the otel config is in the node's config directory?

Would that require us storing all of the meter variables internal to the app structs?

Reply:
this would just require us to initialize telemetry somewhere after the clientContext has been built with all the flags, envs, tomls, etc. 

perhaps if we go this way, we can just chuck a default otel config in the `home` as part of `appd init`?